packages/phoenix-evals/src/phoenix/evals/preview/evaluators.py

```diff
@@ -0,0 +1,615 @@
+            labels=self.labels,
+            include_explanation=self.include_explanation,
+            method=method,
+        )
+        label = response["label"]  # type: ignore
+        explanation = response.get("explanation", None)  # type: ignore
+
+        # Validate that the returned label is one of the valid choices
+        valid_labels = (
+            list(self.labels) if isinstance(self.labels, list) else list(self.labels.keys())
+        )
+        if label not in valid_labels:
+            raise ValueError(
+                f"ClassificationEvaluator '{self.name}' received invalid label '{label}'. "
+                f"Valid labels are: {valid_labels}. "
+            )
+
+        score = self.label_score_map.get(label) if self.label_score_map else None
+        return [
+            Score(
+                score=score,
+                name=self.name,
+                label=label,
+                explanation=explanation,
+                metadata={"model": self.llm.model},  # could add more metadata here
+                source=self.source,
+                direction=self.direction,
+            )
+        ]
+
+    async def _aevaluate(self, eval_input: EvalInput) -> List[Score]:
+        prompt_filled = self.prompt_template.render(variables=eval_input)
+        method = (
+            ObjectGenerationMethod.TOOL_CALLING
+            if isinstance(self.labels, Dict)
+            else ObjectGenerationMethod.AUTO
+        )
+        response = await self.llm.generate_classification(
+            prompt=prompt_filled,
+            labels=self.labels,
+            include_explanation=self.include_explanation,
+            method=method,
+        )  # type: ignore
+        label = response["label"]
+        explanation = response.get("explanation", None)
+
+        # Validate that the returned label is one of the valid choices
+        valid_labels = (
+            list(self.labels) if isinstance(self.labels, list) else list(self.labels.keys())
+        )
+        if label not in valid_labels:
+            raise ValueError(
+                f"ClassificationEvaluator '{self.name}' received invalid label '{label}'. "
+                f"Valid labels are: {valid_labels}. "
+            )
+
+        score = self.label_score_map.get(label) if self.label_score_map else None
+        return [
+            Score(
+                score=score,
+                name=self.name,
+                label=label,
+                explanation=explanation,
+                metadata={"model": self.llm.model},  # could add more metadata here
+                source=self.source,
+                direction=self.direction,
+            )
+        ]
+
+
+# --- Registry & simple evaluator decorator ---
+_registry: Dict[str, Callable[..., List[Score]]] = {}
+
+
+def list_evaluators() -> List[str]:
+    """
+    Return a list of names of all registered evaluators.
+    """
+    return list(_registry.keys())
+
+
+def simple_evaluator(
```

Comment:
One thought here is we might be able to further generalize what we were talking about yesterday by defining the simplest possible protocol for an evaluator:

A callable that accepts any arguments and returns a Score. As a result this decorator can push the mapping logic into the wrapper instead of changing the call signature